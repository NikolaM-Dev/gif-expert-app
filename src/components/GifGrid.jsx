import { useEffect, useState } from 'react';
import PropTypes from 'prop-types';

import GifGridItem from './GifGridItem';

const GifGrid = ({ category }) => {
  const [gifs, setGifs] = useState([]);

  useEffect(() => {
    getGifs();
  }, []);

  const getGifs = async () => {
    const API_KEY = 'gpUFXO3qzly0cHNaGV8lbGzqdpxklvVH';
    const url = `http://api.giphy.com/v1/gifs/search?q=${encodeURI(
      category
    )}&limit=${10}&api_key=${API_KEY}`;
    const resp = await fetch(url);
    const { data } = await resp.json();

    const gifs = data.map((gif) => ({
      id: gif.id,
      title: gif.title,
      url: gif.images?.downsized_medium.url,
    }));

    setGifs(gifs);
  };

  return (
    <>
      <h2>{category}</h2>
      <div className="card__grid">
        {gifs.map((gif) => (
          <GifGridItem key={gif.id} gif={gif} />
        ))}
      </div>
    </>
  );
};

GifGrid.propTypes = {
  category: PropTypes.string.isRequired,
};

export default GifGrid;
