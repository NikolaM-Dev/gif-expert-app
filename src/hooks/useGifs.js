import { useEffect, useState } from 'react';

import getGifs from 'services/getGifs';

const useGifs = ({ category }) => {
  const [state, setState] = useState({
    gifs: [],
    loading: true,
  });

  useEffect(() => {
    getGifs({ category }).then((gifs) => setState({ gifs, loading: false }));
  }, [category]);

  return state;
};

export default useGifs;
