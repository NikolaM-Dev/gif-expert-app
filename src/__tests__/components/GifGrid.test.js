import '@testing-library/jest-dom';
import { shallow } from 'enzyme';

import GifGrid from 'components/GifGrid';
import useGifs from 'hooks/useGifs';
jest.mock('hooks/useGifs');

describe('Test on <GifGrid />', () => {
  const category = 'Dragon Ball';

  it('should show <GifGrid /> correctly', () => {
    useGifs.mockReturnValue({
      gifs: [],
      loading: true,
    });
    const wrapper = shallow(<GifGrid category={category} />);

    expect(wrapper).toMatchSnapshot();
  });

  it('should show items when images are loaded with useGifs', () => {
    const gifs = [
      {
        id: '01',
        title: 'Goku',
        url: 'https://ghipy/goku.gif',
      },
      {
        id: '02',
        title: 'Vegeta',
        url: 'https://ghipy/vegeta.gif',
      },
    ];

    useGifs.mockReturnValue({
      gifs,
      loading: false,
    });
    const wrapper = shallow(<GifGrid category={category} />);

    expect(wrapper.find('p').exists()).toBeFalsy();
    expect(wrapper.find('GifGridItem')).toHaveLength(gifs.length);
    expect(wrapper).toMatchSnapshot();
  });
});
