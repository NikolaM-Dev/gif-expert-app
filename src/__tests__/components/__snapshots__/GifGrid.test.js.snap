// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test on <GifGrid /> should show <GifGrid /> correctly 1`] = `
<Fragment>
  <h2
    className="animate__animated animate__fadeIn"
  >
    Dragon Ball
  </h2>
  <p
    className="animate__animated animate__flash"
  >
    loading
  </p>
  <div
    className="card__grid"
  />
</Fragment>
`;

exports[`Test on <GifGrid /> should show items when images are loaded with useGifs 1`] = `
<Fragment>
  <h2
    className="animate__animated animate__fadeIn"
  >
    Dragon Ball
  </h2>
  <div
    className="card__grid"
  >
    <GifGridItem
      gif={
        Object {
          "id": "01",
          "title": "Goku",
          "url": "https://ghipy/goku.gif",
        }
      }
      key="01"
    />
    <GifGridItem
      gif={
        Object {
          "id": "02",
          "title": "Vegeta",
          "url": "https://ghipy/vegeta.gif",
        }
      }
      key="02"
    />
  </div>
</Fragment>
`;
